src/Microsoft.AspNetCore.Razor.Language/Legacy/MarkupParser.cs

```diff
@@ -0,0 +1,916 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language.Syntax.InternalSyntax;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy
+{
+    internal partial class HtmlMarkupParser : TokenizerBackedParser<HtmlTokenizer>
+    {
+        private const string ScriptTagName = "script";
+
+        private static readonly char[] ValidAfterTypeAttributeNameCharacters = { ' ', '\t', '\r', '\n', '\f', '=' };
+        private static readonly SyntaxToken[] nonAllowedHtmlCommentEnding = new[]
+        {
+            SyntaxFactory.Token(SyntaxKind.Text, "-"),
+            SyntaxFactory.Token(SyntaxKind.Bang, "!"),
+            SyntaxFactory.Token(SyntaxKind.OpenAngle, "<"),
+        };
+
+        //From http://dev.w3.org/html5/spec/Overview.html#elements-0
+        private readonly ISet<string> _voidElements = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
I think it's worth moving these out into another (static) class. I know the Blazor code has a few copies of this set also...